projects/js-packages/svelte-data-sync-client/src/SyncedStore.ts

```diff
@@ -68,22 +69,22 @@ export class SyncedStore< T > {
 
 	/**
 	 * A callback that will synchronize the store in some way.
-	 * By default, this is set to endpoint.POST in the client initializer
+	 * By default, this is set to endpoint.SET in the client initializer
 	 */
-	private setCallback( callback: SyncedStoreCallback< T > ) {
-		this.updateCallback = callback;
+	private setSyncAction( callback: SyncedStoreCallback< T > ) {
+		this.syncAction = callback;
 	}
 
 	/**
 	 * Attempt to synchronize the store with the API.
 	 */
-	private async synchronize( value: T ): Promise< T | ApiError > {
-		if ( ! this.updateCallback ) {
+	private async synchronize( prevValue: T, value: T ): Promise< T | ApiError > {
+		if ( ! this.syncAction ) {
 			return value;
 		}
 
 		try {
-			const result = await this.updateCallback( value, this.abortController.signal );
+			const result = await this.syncAction( prevValue, value, this.abortController.signal );
 
 			// Success is only when the updateCallback result matches the value.
 			if ( this.equals( result, value ) ) {
```

Comment:
This creates weird issues when the value changes in the backend.

i.e. When you are changing module status in multiple tabs, it will condemn one of them to fail.

Reply:
Fixed